### PR TITLE
Use GDI Permission check for dependencies

### LINKED ISSUE #66: [Dependency] Permissions
Currently there is a permissions check for the requested element only. 
But the according dependencies have to be checked for the `listing` permission too, see existing implementation: https://github.com/pimcore/pimcore/blob/e9c9e2d43e56502f000a38c59eb7f92c6029c347/models/Element/Service.php#L192

The idea is to have an endpoint in `generic-data-index-bundle` that can check permissions for more than one object, thus providing a fast way to check permissions on a set of elements.

### PR DESCRIPTION
**Pull request by mattamon**
## Changes in this pull request
Resolves #66

### CODE REVIEW

File: src/Dependency/Hydrator/DependencyHydrator.php [hunk 1]
```diff
@@ -16,53 +16,25 @@
...
-    use ElementProviderTrait;
-
-    public function __construct(
-        private ServiceResolverInterface $serviceResolver,
-    ) {
-
-    }
-
-    public function hydrate(array $dependency): ?Dependency
+    public function hydrate(ElementSearchResultItemInterface $dependency): ?Dependency
```
Comment (martineiber) [inline src/Dependency/Hydrator/DependencyHydrator.php]:
```suggestion
    public function hydrate(ElementSearchResultItemInterface $dependency): Dependency
```

nullable can be removed?

File: src/Dependency/Repository/DependencyRepository.php [hunk 2]
```diff
@@ -27,53 +37,54 @@
...
+    ): ElementSearchResult
     {
-        return $this->getElement(
-            $this->serviceResolver,
-            $elementType,
-            $elementId
-        )
-            ->getDependencies()
-            ->getRequires();
+        $search = $this->searchProvider->createAssetSearch();
```
Comment (martineiber) [inline src/Dependency/Repository/DependencyRepository.php]: Just for my information, the data from the SearchProvider is coming from the Search Server, and we treat it as if it were a DB. Is this why it is in a Repository?

**Review by martineiber — CHANGES_REQUESTED** [review]
Just a small change, Rest LGTM

File: src/Dependency/Repository/DependencyRepository.php (hunk 2 above)
Comment (mattamon) [reply src/Dependency/Repository/DependencyRepository.php]: Yeah kind of. Maybe the name is not fitting, but then I really need to refactor the whole thing and maybe put everything into the service or move it to some other class. 
Not sure thought I just tried to make it work with the GDI

File: src/Dependency/Hydrator/DependencyHydrator.php (hunk 1 above)
Comment (mattamon) [reply src/Dependency/Hydrator/DependencyHydrator.php]: I removed it. Not sure why it was there in the first place. It is like the permissions. Why would we get an empty dependency from the core, or an empty permissions.

**Review by martineiber — APPROVED** [review]
LGTM :+1:

**Pull request merged**